mayastor/src/bdev/nexus/nexus_io.rs

```diff
@@ -338,9 +338,8 @@ impl Bio {
                     nexus.reconfigure(DREvent::ChildFault).await;
                     //nexus.remove_child(&uri).await.unwrap();
                     bdev_destroy(&uri).await.unwrap();
-                    if nexus.status() != NexusStatus::Faulted {
-                        nexus.resume().await.unwrap();
-                    } else {
+                    nexus.resume().await.unwrap();
```

Comment:
Could you please add comments with explanation why it's safe to always unconditionally resume Nexus even if it's (potentially) in Faulted state ? At the first glance eliminating check against NexusStatus::Faulted looks strange, as this state means exactly "no I/O to the Nexus, please":

```
pub enum NexusStatus {
    /// The nexus cannot perform any IO operation
    Faulted,
    /// Degraded, one or more child is missing but IO can still flow
    Degraded,
    /// Online
    Online,
}
```

Reply:
I've updated the code comments for the subsystem pause/resume functions. SPDK considers these to be synchronisation primitives so pause is intended to be temporary.

Reply:
I see, thanks.